chainer/utils/experimental.py

```diff
@@ -0,0 +1,107 @@
+import chainer
+import inspect
+import warnings
+
+
+def experimental(api_name=None):
```

Comment:
What is the appropriate way to declare a specific usage of some existing feature as experimental (e.g., a new argument and a new type of an existing argument)? Do you want to use `experimental` for these cases, or prefer to directly use `warnings.warn`?